internal/controllers/port/actuator.go

```diff
@@ -225,14 +226,20 @@ func newActuator(ctx context.Context, orcObject *orcv1alpha1.Port, controller ge
 	}, nil
 }
 
-func newCreateActuator(ctx context.Context, orcObject *orcv1alpha1.Port, controller generic.ResourceController) ([]generic.ProgressStatus, *portCreateActuator, error) {
+func newCreateActuator(ctx context.Context, orcObject *orcv1alpha1.Port, controller generic.ResourceController) (*portCreateActuator, []generic.ProgressStatus, error) {
+	// Ensure credential secrets exist and have our finalizer
+	_, progressStatus, err := credentialsDependency.GetDependencies(ctx, controller.GetK8sClient(), orcObject, func(*corev1.Secret) bool { return true })
```

Comment:
It's not clear to me why we don't add the same dependency in the delete actuator. Is it on purpose?

Reply:
The finalizer should already have been added on creation, so there's no need.